### PR TITLE
Silee2/external project rpath

### PR DESCRIPTION
**Pull request by silee2**
Set all shared library rpath on linux to "$ORIGIN". All shared libraries are currently placed in a single directory ${NGRAPH_BUILD_DIR}.

### CODE REVIEW

**Comment by silee2** [comment]
Blocking until Linux CI has patchelf installed.

File: CMakeLists.txt [hunk 1]
```diff
@@ -58,12 +58,8 @@ set(NGRAPH_INCLUDE_PATH
...
 
 if (APPLE)
-    # Suppress an OS X-specific warning.
-    if (POLICY CMP0042)
-        cmake_policy(SET CMP0042 NEW)
-    else()
-        set(MACOS_RPATH ON)
-    endif()
+    # Enable MACOS_RPATH by default.
+    cmake_policy(SET CMP0042 NEW)
```
Comment (silee2) [inline CMakeLists.txt]: CMake required version is at least 3.1 so no need to check if POLICY CMP0042 is available.

**Comment by diyessi** [comment]
If we need patchelf, we'll need to update the doc and some other things.

**Comment by silee2** [comment]
@diyessi patchelf is for patching prebuilt libraries. mklml does not have an rpath set. It is assuming libiomp5 is a system library. The whole OpenMP situation is not pretty. libngraph is forced to link against libiomp5 and "-Wl,--no-as-needed" is used to prevent gcc from linking against libgomp. So far, libngraph is loaded before mklml so mklml missing rpath was not an issue as libiomp5 would be already loaded in memory but the whole setup looks fragile.

**Comment by silee2** [comment]
@diyessi We could ask mkl team to set $ORIGIN as rpath for the libmklml bundled with mkl-dnn. Then we don't need patchelf.

**Comment by diyessi** [comment]
It would be good to ask mkl to fix their linking, but, in the mean time, if we have a new build dependency we need to update the build instructions to get the patchelf script (if I remember, it's a simple sed script since it's just replacing a constant's value in the binary).

**Comment by silee2** [comment]
looks like clear linux does not have patchelf. Will have to remove patchelf usage from this PR and add a workaround in a future PR.

**Pull request merged**